Quantization oneDNN backend only support VNNI CPU (#103653)
pytorch/pytorch: docs/source/quantization.rst

you call the `torch.ao.quantization.get_default_qconfig(backend)` or `torch.ao.quantization.get_default_qat_qconfig(backend)` function to get the default ``qconfig`` for
``x86`` or ``qnnpack`` backend

> 2. If ``onednn`` backend is selected, 8 bits for activation will be used in the default qconfig mapping ``torch.ao.quantization.get_default_qconfig_mapping('onednn')``
> and default qconfig ``torch.ao.quantization.get_default_qconfig('onednn')``. It is recommended to be used on CPUs with Vector Neural Network Instruction
> support. Otherwise, setting ``reduce_range`` to True of the activation's observer to get better accuracy on CPUs without Vector Neural Network Instruction
> support.

Reviewer: ```suggestion
2. If ``onednn`` backend is selected, 8 bits for activation will be used in the default qconfig mapping ``torch.ao.quantization.get_default_qconfig_mapping('onednn')``
and default qconfig ``torch.ao.quantization.get_default_qconfig('onednn')``. It is recommended to be used on CPUs with Vector Neural Network Instruction (VNNI)
support. Otherwise, setting ``reduce_range`` to True of the activation's observer to get better accuracy on CPUs without VNNI
support.
```

Author: Thanks and changed.

you call the `torch.ao.quantization.get_default_qconfig(backend)` or `torch.ao.quantization.get_default_qat_qconfig(backend)` function to get the default ``qconfig`` for
``x86`` or ``qnnpack`` backend

> 2. If you are using the ``onednn`` backend, we will use 8 bits for activation in the default qconfig mapping ``torch.ao.quantization.get_default_qconfig_mapping('onednn')``

Reviewer 2: English is not my native, but shouldn't documentation better be in passive tense
```suggestion
2. If ``onednn`` backend is selected, 8 bits for activation in the default qconfig mapping ``torch.ao.quantization.get_default_qconfig_mapping('onednn')`` will be used.
```

Author: Thanks for the suggestions. Update it.

> 2. If you are using the ``onednn`` backend, we will use 8 bits for activation in the default qconfig mapping ``torch.ao.quantization.get_default_qconfig_mapping('onednn')``
> and qconfig ``torch.ao.quantization.get_default_qconfig('onednn')``. It means onednn quantization backend only support CPU with VNNI instruction.

Reviewer: ```suggestion
2. If you are using the ``onednn`` backend, we will use 8 bits for activation in the default qconfig mapping ``torch.ao.quantization.get_default_qconfig_mapping('onednn')``
and qconfig ``torch.ao.quantization.get_default_qconfig('onednn')``. The ``onednn`` quantization backend is recommended to be used on CPUs with Vector Neural Network Instruction support.
```

> and qconfig ``torch.ao.quantization.get_default_qconfig('onednn')``. The ``onednn`` quantization backend is recommended to be used on CPUs with Vector Neural Network Instruction support.

Reviewer 3: "only support VNNI CPU" is much stricter than "recommended to be used on CPUs with Vector Neural Network Instruction support."

If the `onednn` quantization backend only supports processors with VNNI, then the documentation should say that. Also, the code should warn the user when the `onednn` backend is used on a CPU without VNNI or throw an exception.

Author: Thanks for the comment. 
I am working on to add the warning, will keep you updated.
